include/vcpkg/base/cache.h

```diff
@@ -49,4 +50,32 @@ namespace vcpkg
     private:
         mutable std::map<Key, Value, Compare> m_cache;
     };
+
+    template<class Key, class Value, class Compare = std::less<>>
+    struct AtomicCache
+    {
+        template<class KeyIsh,
+                 class F,
+                 std::enable_if_t<std::is_constructible<Key, const KeyIsh&>::value &&
+                                      detail::is_callable<Compare&, const Key&, const KeyIsh&>::value,
+                                  int> = 0>
+        const Value& get_lazy(const KeyIsh& k, F&& f) const
+        {
+            auto it = m_cache.lower_bound(k);
```

Comment:
This is a data race, since nothing stops `emplace_hint` from being called by another thread at the same time.